packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -3471,12 +3471,49 @@ namespace Tpetra {
     TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC(
       ! hasColMap () || getColMap ().is_null (), std::runtime_error,
       "The graph must have a column Map before you may call this method.");
+    LocalOrdinal numLocalRows = this->getNodeNumRows ();
     TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC(
-      static_cast<size_t> (rowPointers.size ()) != this->getNodeNumRows () + 1,
+      static_cast<LocalOrdinal> (rowPointers.size ()) != numLocalRows + 1,
       std::runtime_error, "rowPointers.size() = " << rowPointers.size () <<
-      " != this->getNodeNumRows()+1 = " << (this->getNodeNumRows () + 1) <<
+      " != this->getNodeNumRows()+1 = " << (numLocalRows + 1) <<
       ".");
 
+#ifdef HAVE_TPETRA_DEBUG
```

Comment:
Please use `Tpetra::Details::Behavior::debug()` for a run-time test.  This will let users decide at run time if they want to do debug checks, even in a release build.  On some platforms, it's impossible or infeasible to rebuild or to finish a debug build.

Reply:
```c++
#include "Tpetra_Details_Behavior.hpp"
// ...
const bool debug = Tpetra::Details::Behavior::debug();
if (debug) {
  // ... do expensive debug checks ...
}
```